### PR TITLE
feat(editor): Implement new app layout

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
## Summary
Implementing new app layout that will enable AI Assistant sidebar to render alongside the rest of the app. This also moves all modals to the app grid but leaves the option to still teleport them to the `body` if we want them to overlay everything else.
See [Linear ticket](https://linear.app/n8n/issue/ADO2-62/rework-modals-to-fit-with-the-assistant) for details.

[Latest E2E run](https://github.com/n8n-io/n8n/actions/runs/10595843053)

## Related Linear tickets, Github issues, and Community forum posts
Closes ADO2-62

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/components/NDVDraggablePanels.vue [hunk 1]
```diff
@@ -107,6 +109,27 @@ watch(windowWidth, (width) => {
...
+	} else {
+		// It looks the best if we delay closing adjustment until slide transition is done
+		setTimeout(() => {
+			containerWidth.value = window.innerWidth;
+		}, 200);
+	}
+});
+
+// As assistant sidebar width changes, recalculate the total width regularly
+watch(assistantSidebarWidth, async () => {
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/NDVDraggablePanels.vue:129]: Nit: no need for this to be async

File: packages/editor-ui/src/components/TextEdit.vue [hunk 2]
```diff
@@ -60,7 +60,7 @@ const closeDialog = () => {
 	<div v-if="dialogVisible">
 		<el-dialog
 			:model-value="dialogVisible"
-			append-to-body
+			append-to="#app-modals"
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/TextEdit.vue:63]: We seem to be using this ID in several places, perhaps some constant would make sense to have a single source of truth

File: packages/editor-ui/src/components/NDVDraggablePanels.vue [hunk 3]
```diff
@@ -357,15 +380,18 @@ function onDragEnd() {
...
+			windowWidth: containerWidth.value,
 			position: mainPanelDimensions.value.relativeLeft,
 		});
 	}, 0);
 	storePositionData();
 }
 
 function setTotalWidth() {
-	windowWidth.value = window.innerWidth;
+	const appModals = document.getElementById('app-grid');
```
[the remark this answers is not in the document]
Comment (MiloradFilipovic) [reply packages/editor-ui/src/components/NDVDraggablePanels.vue:391]: Yeah, that makes sense since the element is part of the `App` component

**Review by OlegIvaniv — DISMISSED** [review]
Looks good!

**Review by OlegIvaniv — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.58.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.58.0)

**Pull request merged**